### PR TITLE
adding reporting for failed encodings

### PR DESCRIPTION
**Pull request by b-maldoca**

### CODE REVIEW

File: maldoca/base/encoding_error.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef MALDOCA_CHROME
+
+namespace maldoca {
+namespace {
```
Comment (dtao-oss) [inline maldoca/base/encoding_error.h:18]: https://g3doc.corp.google.com/devtools/cymbal/clang_tidy/g3doc/checks/google-build-namespaces.md?cl=head  not supposed to use anonymous namespace in header.

File: maldoca/service/common/process_doc.cc [hunk 2]
```diff
@@ -299,7 +300,17 @@ absl::Status DocProcessor::ProcessDoc(absl::string_view file_name,
     }
   }
 
-  MALDOCA_RETURN_IF_ERROR(current_document_pipeline->Process(doc));
+  ::maldoca::ResetFailedEncoding();
```
Comment (dtao-oss) [inline maldoca/service/common/process_doc.cc:303]: maybe we should do this in BufferToUtf8::Init() so it's always referring to the last try.

**Review by dtao-oss — CHANGES_REQUESTED** [review]

File: maldoca/base/encoding_error.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+static std::string& failed_encoding() {
+  thread_local std::string failed_encoding("");
```
[the remark this answers is not in the document]
Comment (b-maldoca) [reply maldoca/base/encoding_error.h:21]: done

File: maldoca/service/common/process_doc.cc (hunk 2 above)
Comment (b-maldoca) [reply maldoca/service/common/process_doc.cc:303]: done

File: maldoca/base/encoding_error.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef MALDOCA_CHROME
+
+namespace maldoca {
+namespace {
+
+static std::string& failed_encoding() {
```
[the remark this answers is not in the document]
Comment (b-maldoca) [reply maldoca/base/encoding_error.h]: tried to make it explicit that it's basically a global var but changed to a func (due to the static init issue). anyways, changed both.

File: maldoca/base/encoding_error.h [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  thread_local std::string failed_encoding("");
+  return failed_encoding;
+}
+
+[[maybe_unused]] void SetFailedEncoding(const char* encoding) {
+  failed_encoding() = std::string(encoding);
+}
+
+[[maybe_unused]] std::string* GetFailedEncoding() { return &failed_encoding(); }
```
[the remark this answers is not in the document]
Comment (b-maldoca) [reply maldoca/base/encoding_error.h]: removed

**Review by dtao-oss — APPROVED** [review]

**Pull request merged**